### PR TITLE
Convert Filebeat mysql.* to ECS

### PR DESCRIPTION
**Pull request by webmat**
**Caveats**

- Got rid of workaround for an Ingest Node issue fixed in 5.0, in 2016

**Renames**

- mysql.error.message => message
- mysql.error.level => log.level
- mysql.error.thread_id => process.thread.id
- mysql.slowlog.ip => source.ip
- mysql.slowlog.host => source.domain
- mysql.slowlog.user => user.name

**TODO**

- [x] Add `-expected.json` log for the error logs
- [x] Get rid of workaround for https://github.com/elastic/elasticsearch/issues/22117, which was fixed in 5.0
- [x] Set event.created
- [x] Coerce int fields: process.thread.id, mysql.slowlog.id, mysql.slowlog.rows_sent, mysql.slowlog.rows_examined and mysql.slowlog.timestamp
- [x] Coerce float fields: mysql.slowlog.query_time.sec, mysql.slowlog.lock_time.sec
- [x] Populate event.duration based on mysql.slowlog.query_time.sec
- [x] Alias renamed fields to their ECS counterpart, not forgetting `migration: true`
- [x] Document field migrations in ecs-migration.yml
- [x] Changelog

### CODE REVIEW

File: filebeat/module/mysql/slowlog/test/mysql-5.7.22.log [hunk 1]
```diff
@@ -9,7 +9,7 @@ select sleep(15);
 SET timestamp=1533630467;
 SELECT count(*) FROM mysql.user WHERE user='root' and password='';
 # Time: 2018-08-07T16:27:47.169604+08:00
-# User@Host: apphost[apphost] @ apphost [1.1.1.1]  Id: 10997316
+# User@Host: appuser[appuser] @ apphost [1.1.1.1]  Id: 10997316
```
Comment (ruflin) [inline filebeat/module/mysql/slowlog/test/mysql-5.7.22.log]: Should not make a difference but I wonder why this was changed?

File: filebeat/module/mysql/slowlog/test/mysql-5.7.22.log-expected.json [hunk 2]
```diff
@@ -3,85 +3,89 @@
...
-        "mysql.slowlog.host": "localhost",
-        "mysql.slowlog.lock_time.sec": "0.000061",
+        "mysql.slowlog.lock_time.sec": 6.1e-05,
         "mysql.slowlog.query": "SELECT count(*) FROM mysql.user WHERE user='root' and password='';",
-        "mysql.slowlog.query_time.sec": "0.000153",
-        "mysql.slowlog.rows_examined": "5",
-        "mysql.slowlog.rows_sent": "1",
-        "mysql.slowlog.timestamp": "1533630467",
-        "mysql.slowlog.user": "debian-sys-maint"
+        "mysql.slowlog.query_time.sec": 0.000153,
```
Comment (ruflin) [inline filebeat/module/mysql/slowlog/test/mysql-5.7.22.log-expected.json]: Not for this PR but it's kind of add that we have `.sec` but then a micro second unit.

Comment (webmat) [reply filebeat/module/mysql/slowlog/test/mysql-5.7.22.log-expected.json]: Yeah. Well the unit is seconds, but there's more precision. So meh :-)

**Comment by webmat** [comment]
jenkins, test this

**Pull request merged**